src/ui/utils/hooks/useDynamic.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useState } from 'react';
+import type { FunctionComponent } from 'react';
+import type { IconName } from '../vite-svgr/types/IProps';
+
+export function useDynamic(iconName: IconName) {
```

Comment:
The name of this function is ambiguous. Please choose a more descriptive name to improve clarity and maintainability.